elastic/kibana
# [Fleet] Adjustments to the "Add agent" flyout

## Summary

When adding an agent in certain contexts we already have a specific policy we'd like to apply. In those cases the "Add agent" flyout should not prompt the user with a list of policies to choose from but rather just use the policy from the context.

## How to test

1. Start Kibana + ES on basic
2. Follow the in-app instructions for setting up a fleet server where agents can be registered
3. To see the "Add agent" flyout with the policy selection step go to Fleet > Agents > "+ Add Agent"
4. To see the "Add agent" flyout with a policy from the context go to Fleet > Policies > "..." on a specific policy row in the table > "+ Add agent"

All locations where the "Add agent" flyout will have a policy from the context are:

Policy list view:
<img width="1406" alt="Screenshot 2021-06-07 at 16 09 18" src="https://user-images.githubusercontent.com/8155004/121176611-b2f6d780-c85c-11eb-900b-b32bfa42fdf7.png">

Editing a specific policy:
<img width="1372" alt="Screenshot 2021-06-07 at 16 09 11" src="https://user-images.githubusercontent.com/8155004/121176643-bd18d600-c85c-11eb-9c24-8e2f7d78bb53.png">


## Screenshot of changes

Adjusted the "Add agent" flyout footer since there is no resulting action in the flyout, the "Continue" button looked like a call to action.

<img width="840" alt="Screenshot 2021-06-07 at 14 12 55" src="https://user-images.githubusercontent.com/8155004/121174061-cfdddb80-c859-11eb-9f94-3ecf75629895.png">

### Comparison of with and without specific policy add agent flyout

#### Without specific policy (currently used everywhere)

| Managed instructions | Standalone instructions |
|-----------------------|------------------ |
|  <img width="848" alt="Screenshot 2021-06-08 at 13 12 58" src="https://user-images.githubusercontent.com/8155004/121175846-d40af880-c85b-11eb-978a-2917016980cc.png"> |  <img width="832" alt="Screenshot 2021-06-08 at 13 13 05" src="https://user-images.githubusercontent.com/8155004/121175887-e2591480-c85b-11eb-8d3c-96d349d8fdc3.png"> |


#### <a name="withspecificpolicy" />With specific policy (new)

| Managed instructions | Standalone instructions |
|-----------------------|------------------ |
| <img width="839" alt="Screenshot 2021-06-08 at 13 13 18" src="https://user-images.githubusercontent.com/8155004/121176084-18969400-c85c-11eb-9fdd-0c6027843bc1.png">  |  <img width="843" alt="Screenshot 2021-06-08 at 13 13 26" src="https://user-images.githubusercontent.com/8155004/121176118-23512900-c85c-11eb-9451-b5730d86f0b7.png"> |

## Release note

We updated the "Add agent" flyout to use the policy in the context where the flyout was opened or show a list of policies if there is no specific policy in the current UI context.


### Checklist

Delete any items that are not applicable to this PR.
